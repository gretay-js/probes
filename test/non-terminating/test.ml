(* simple example: run forever, print something once, when a probe is hit for
   the first time. *)

let p1_hit = ref false

let p2_hit = ref false

let p3_hit = ref false

let on_p name hit =
  if not !hit then (
    hit := true;
    Printf.printf "Hit %s\n" name;
    flush stdout
  )

let () =
  while true do
    [%probe "p1" (on_p "p1" p1_hit)];
    [%probe "p2" (on_p "p2" p2_hit)];
    [%probe "p3" (on_p "p3" p3_hit)];
  done
