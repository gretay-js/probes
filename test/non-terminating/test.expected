0:p1
1:p2
p1 disabled
p2 disabled
Test p1
p1 disabled
p2 enabled
p1 disabled
p2 disabled
Test p2
p1 enabled
p2 enabled
p1 disabled
p2 disabled
p1 enabled
p2 enabled
p1 disabled
p2 disabled
Tracing test.exe with process id 12272 failed. Process 12272 killed with signal -7.

